### PR TITLE
[FEATURE] SparkDFExecutionEngine can load data accessed by ConfiguredAssetAzureDataConnector (integration tests are included).

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-192/GREAT-243
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/datasource/data_connector/data_connector.py
Comment (cdkini) [inline great_expectations/datasource/data_connector/data_connector.py:110]: If we're leaving this TODO, could we please add the Jira ticket here?

File: great_expectations/datasource/data_connector/inferred_asset_file_path_data_connector.py
Comment (cdkini) [inline great_expectations/datasource/data_connector/inferred_asset_file_path_data_connector.py:109]: Nice catch!

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py
Comment (cdkini) [inline tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py]: I don't feel particularly strongly about this but why the change from `AZURE_CREDENTIAL` to `credential`? Happy to stick with whatever you think works better here but would love to hear the reasoning for my own edification.

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py
Comment (cdkini) [inline tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py]: Could we obfuscate these with `<YOUR_X_HERE>`? I don't know if I did it for the other example but it would make it more applicable to users when we eventually write the docs.

Could also push that off to when we write docs I'll leave it up to you.

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_yaml_example.py
Comment (cdkini) [inline tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_yaml_example.py:26]: Same note as above. We do this here so perhaps let's make it consistent throughout?

File: great_expectations/execution_engine/sparkdf_execution_engine.py
Comment (cdkini) [inline great_expectations/execution_engine/sparkdf_execution_engine.py:262]: Could we write a test where we hit this? Or is it not possible because we mock? Just want to ensure complete test coverage

File: great_expectations/execution_engine/sparkdf_execution_engine.py
Comment (cdkini) [inline great_expectations/execution_engine/sparkdf_execution_engine.py:178]: Very nitpicky but we could probably consolidate to one line? I'll defer to your judgement.

**Review by cdkini — COMMENTED** [review]
Very minor comments. Please take a look and let me know your thoughts.

Happy to accept once we are on the same page with my comments 😄

File: great_expectations/datasource/data_connector/inferred_asset_file_path_data_connector.py
Comment (cdkini) [reply great_expectations/datasource/data_connector/inferred_asset_file_path_data_connector.py:109]: And all the subsequent catches :)

File: great_expectations/datasource/data_connector/data_connector.py
Comment (cdkini) [reply great_expectations/datasource/data_connector/data_connector.py:110]: Unless we have some aversion to writing in Jira refs in the codebase? I could go either way here.

Comment (alexsherstinsky) [reply great_expectations/datasource/data_connector/data_connector.py:110]: @cdkini From what I recall, we are not supposed to put the internal Atlassian instance's tickets into the OSS repository...

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py
Comment (alexsherstinsky) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py]: @cdkini Stylistically, I did not want constants to be in this form.  If you think that this is not proper, I can revert uniformly.  Please let me know what your preference is.  Thank you.

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py
Comment (alexsherstinsky) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py:29]: @cdkini I followed the pattern already in place.  If you agree that this is so, how about we address at a later date?  Thanks!

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py
Comment (cdkini) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py]: what is your preferred alternative for constants? Just asking since it is a convention I've used for a while (but may not necessarily be idiomatic Python).

Let's do what you think is best here but I'd love to get your thoughts just for better understanding 🐍

File: great_expectations/datasource/data_connector/data_connector.py
Comment (cdkini) [reply great_expectations/datasource/data_connector/data_connector.py:110]: That's fair please disregard this!

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_yaml_example.py
Comment (alexsherstinsky) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_yaml_example.py:26]: @cdkini This is YAML -- I believe we treat them differently.

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py
Comment (alexsherstinsky) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py]: @cdkini This is Python -- so we adhere to the explicit pattern, since an easy string substitution is not available.

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_yaml_example.py
Comment (cdkini) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_yaml_example.py:26]: I don't believe that is the case with S3 and GCS actually. I could be wrong but I think the less specificity the better here (just to help the future doc writer)

File: great_expectations/execution_engine/sparkdf_execution_engine.py
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sparkdf_execution_engine.py:178]: @cdkini I did this on purpose (to follow what we do with `spark_config`).

Comment (cdkini) [reply great_expectations/execution_engine/sparkdf_execution_engine.py:178]: okay then please disregard! resolving

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py
Comment (cdkini) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py]: Should be a dictionary update no? I'm quite confident our approach is substitution in all other connectors so I'd like to push on this. Let me know if you want to discuss.

File: great_expectations/execution_engine/sparkdf_execution_engine.py
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sparkdf_execution_engine.py:262]: @cdkini Not sure right now.  I would have to "soft-corrupt `PySpark`" to remove the reader method (e.g., `.csv()`.  Perhaps for a separate PR just focused on adding this test, which we have not had thus far?

**Review by cdkini — APPROVED** [review]
Approving per our conversation. Please make the minor changes before merging.

Thanks for all the hard work as always 🎉

**Comment by alexsherstinsky** [comment]
@cdkini -- changes have been made and committed!  Thank you for taking the time to review and discuss!

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py
Comment (alexsherstinsky) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/pandas/configured_python_example.py]: Reverted to Constant.

File: tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py
Comment (alexsherstinsky) [reply tests/integration/docusaurus/connecting_to_your_data/cloud/azure/spark/configured_python_example.py]: Done.

**Pull request merged**
